src/AlchemistV2.sol

```diff
@@ -527,9 +529,17 @@ contract AlchemistV2 is IAlchemistV2, Initializable, Multicall, Mutex {
             revert UnsupportedToken(rewardToken);
         }
 
-        TokenUtils.safeTransfer(rewardToken, admin, amount);
+        // If VSP Reward
+        if (rewardToken == 0x1b40183EFB4Dd766f11bDa7A7c3AD8982e998421) {
```

Comment:
having this be so esper-specific in the alchemist is a bit weird.

lets use a new contract to encapsulate this functionality. we can delegateCall to it so that the alchemist is the one that VesperRewards sees as `msg.sender`

interface can be something like
```
interface IRewardClaimer {
    function claim(address yieldToken);
}
```

that way if we end up having more reward contracts like vesper we can build more XRewardClaimer contracts and use them generically in the same way.
here, the alchemist would accept the address of the reward claimer (sent from the keeper) and just call `IRewardClaimer(rewardClaimer).claim(yieldToken)`, then continue from line 536.

Reply:
oh even better, instead of putting the `claim()` function in a new contract, just put it in the RewardCollector.  alchemist can then call `IRewardCollector(msg.sender).delegateCall(abi.encodeWithSignature("claim(address)", yieldToken)`